dotnet/src/SemanticKernel.Core/Functions/SemanticFunction.cs

```diff
@@ -72,9 +72,12 @@ internal sealed class SemanticFunction : ISKFunction, IDisposable
             description: promptTemplateConfig.Description,
             pluginName: pluginName,
             functionName: functionName,
+            serviceSelector: serviceSelector,
             loggerFactory: loggerFactory
-        );
-        func.SetAIConfiguration(promptTemplateConfig.GetDefaultRequestSettings());
+        )
+        {
+            _modelSettings = promptTemplateConfig.ModelSettings
```

Comment:
Why does a PromptTemplateConfiguration has model settings ? This configuration should be passed directly to the `FromSemanticConfig()`.

PromptTemplateEngine, should not have any logic or knowledge of the `Model` settings.

Reply:
`PromptTemplateConfig.cs` is the model object for a config.json. We allow the model settings to be set in the config.json.